lib/Connection.js

```diff
@@ -123,10 +124,34 @@ class Connection {
       callback(new RequestAbortedError(), null)
     }
 
+    const onSocket = socket => {
+      /* istanbul ignore else */
+      if (!socket.isSessionReused()) {
+        socket.once('secureConnect', () => {
+          // Check if certificate is valid
+          if (socket.authorized === false) {
+            onError(new Error(socket.authorizationError))
+            request.once('error', () => {}) // we need to catch the request aborted error
+            return request.abort()
+          }
```

Comment:
I’m not entirely sure this is needed, as this check would catch the self-signed certificate.
One of the test is failing because of this check, if we remove it, everything works as expected.

Reply:
I suspect it's not needed since we're going to validate a fingerprint of the top-most CA certificate, that will most-likely be a self-signed one (from the tech spec `The fingerprint (SHA256) of the CA certificate that is used to sign the certificate that the Elasticsearch node presents for TLS. `). @jkakavas is my understanding correct?

Reply:
> @jkakavas is my understanding correct?

Yes. We elected to use the CA certificate for the pinning as it originally felt like it provided for more options. It _will_ be a self signed CA certificate that Elasticsearch generates on startup